clang/test/CheckedC/inferred-bounds/multiple-widened-bounds.c

```diff
@@ -0,0 +1,568 @@
+// Tests for datafow analysis for bounds multiple widening of _Nt_array_ptr's.
+//
+// RUN: %clang_cc1 -fdump-widened-bounds -verify -verify-ignore-unexpected=note -verify-ignore-unexpected=warning %s | FileCheck %s
```

Comment:
Can you also add the following tests:

1.  Tests for out-of-order dereferences, like:
`if (*(p + 2) && *p  && *(p + 1))`

2. Tests for expressions with parentheses, like:
`if (((*p && *(p + 1)) && *(p + 2)) && (*(p + 3) && *(p + 4)))`

3. Tests for OR, like:
`if (*p || *(p + 1))`
In this case, there should be no bounds widening.

Reply:
for case 1 I think f19 covers the out of order dereferences.

Reply:
Yes, I see it now. Case 1 seems to be already covered.